services/motion/builtin/move_request.go

```diff
@@ -18,11 +18,19 @@ import (
 	"go.viam.com/rdk/spatialmath"
 )
 
+const (
+	defaultReplanCostFactor = 1.0
+	defaultMaxReplans       = -1 // Values below zero will replan infinitely
+)
+
 // moveRequest is a structure that contains all the information necessary for to make a move call.
 type moveRequest struct {
-	config        *motion.MotionConfiguration
-	planRequest   *motionplan.PlanRequest
-	kinematicBase kinematicbase.KinematicBase
+	config           *motion.MotionConfiguration
+	planRequest      *motionplan.PlanRequest
+	seedPlan         motionplan.Plan
+	kinematicBase    kinematicbase.KinematicBase
+	maxReplans       int
```

Comment:
I still dont love adding maxReplans here.  Its not in scope yet and test convenience should not dictate the structure of the code we write.  

The reason I'm bringing this up again and in this location is that the MotionConfiguration is intended to be the one stop shop for things a user can configure and as such we shouldnt be adding additional members to this struct.   To this end I think we should move replanCostFactor inside this struct as it is technically still configurable by the user.  And I don't think we should be adding other backdoors here that are not configurable in the way we want users of the service to use it.

Reply:
I don't like that our failure to scope the absence of an infinite loop, means that it is a good idea to implement an infinite loop. IMO if it wasn't scoped one way or another, it should fall back to engineering best practices, which is usually "don't loop infinitely".

The `MotionConfiguration` is intended to be a 1-to-1 match with things that we have put into our API specifically, so that's not the right place for that (or for replanCostFactor, which is scoped to have us pick a default and bake that in; not to put into our API and enshrine as user-configurable).

This is a private member of a private struct and specifies behavior only relevant to this struct, so I think this is the right place for both parameters.

I see the point about poll rate, but I think we will still want some value here.

Reply:
I thought we talked in our meeting yesterday about making these changes? Was this comment written before or after that?

Reply:
That comment was written ~15 minutes before our meeting